### PR TITLE
Improve AdminSearchController#find_students.

### PR DESCRIPTION
**Pull request by ashercodeorg**
This makes several improvements and fixes:
1. Fixes the ordering of the column headers in the view to match the data.
1. Resolves a TODO to show a warning if multiple teachers match.
1. Displays an alert to the admin if the section does not exist or is deleted.
In doing so, resolves this [HB](https://app.honeybadger.io/projects/3240/faults/34359430) error (namely `NoMethodError: undefined method 'students' for nil:NilClass`) happening when the searched section does not exist.

EXAMPLE WARNING (section not found):

image

### CODE REVIEW

**Comment by ashercodeorg** [comment]
FYI @Hamms.

File: dashboard/test/controllers/admin_search_controller_test.rb [hunk 1]
```diff
@@ -17,6 +17,40 @@ class AdminSearchControllerTest < ActionController::TestCase
...
+  # find_student tests
+  #
+
+  test 'find_students flashes warning if multiple teachers match' do
+    create_list :teacher, 2, name: 'multiple'
+
+    get :find_students, params: {teacherNameFilter: 'multiple'}
+
+    assert_response :success
+    assert_select '.container .alert-danger', 'Multiple teachers matched the name and email search criteria.'
```
Comment (ashercodeorg) [inline dashboard/test/controllers/admin_search_controller_test.rb:30]: Note that I don't understand why the `assert_equal message, flash[:alert]` syntax fails.

Comment (Hamms) [reply dashboard/test/controllers/admin_search_controller_test.rb:30]: Intersting also that it's (presumably) working for the `undelete_section` tests but not for the `lookup_section` test

Comment (ashercodeorg) [reply dashboard/test/controllers/admin_search_controller_test.rb:30]: Yeah, I noticed. Honestly, I didn't spend almost any time trying to understand, it didn't seem worth it.

**Comment by ashercodeorg** [comment]
@Hamms: This LG?

**Pull request merged**